.github/workflows/generic-deb-workflow.yml

```diff
@@ -88,64 +109,111 @@ jobs:
 
     - name: Raise App version
+      if: ${{ inputs.deploy == 'true' }}
       uses: CasperWA/push-protected@v2.14.0
       with:
         token: ${{ secrets.auto_commit_pwd }}
         branch: ${{ steps.pre_raise.outputs.branch }}
         unprotect_reviews: true
 
-    - name: Publish to Nexus ppa-testing
-      run: |
-          NEXUS_ENDPOINT="artifacts.cloud.mov.ai"
-          NEXUS_REPO="ppa-testing"
-          FILE_LOCATION=$(find artifacts/*.deb)
-          curl -u "${{ secrets.nexus_publisher_user }}:${{ secrets.nexus_publisher_password }}" \
-          -H "Content-Type: multipart/form-data" \
-          --data-binary "@$FILE_LOCATION" \
-          "https://$NEXUS_ENDPOINT/repository/$NEXUS_REPO/"
-    - name: Publish to Nexus ppa-main
-      run: |
-          NEXUS_ENDPOINT="artifacts.cloud.mov.ai"
-          NEXUS_REPO="ppa-main"
-          FILE_LOCATION=$(find artifacts/*.deb)
-          curl -u "${{ secrets.nexus_publisher_user }}:${{ secrets.nexus_publisher_password }}" \
-          -H "Content-Type: multipart/form-data" \
-          --data-binary "@$FILE_LOCATION" \
-          "https://$NEXUS_ENDPOINT/repository/$NEXUS_REPO/"
+    - name: Upload artifacts
+      uses: actions/upload-artifact@v4
+      with:
+        name: packages
+        path: artifacts/*
+
     - name: Commit info
+      if: ${{ inputs.deploy == 'true' }}
       id: commit
       shell: bash
       run: |
         commit_hash=$(git log --format="%H" -n 1)
         echo "commit_id=$commit_hash" >> $GITHUB_OUTPUT
 
     - name: Create Github Release
-      shell: bash
-      run: |
-        product_version=${{ steps.vars-after-raise.outputs.pkg_version }}
-        previous_version_option=""
-        if gh release view ${{ steps.vars.outputs.pkg_version }} &>/dev/null; then
-          previous_version_option="--notes-start-tag ${{ steps.vars.outputs.pkg_version }}"
-        fi
-        gh release create -p --generate-notes $previous_version_option --target $commit_hash -t "Release of $product_version" $product_version
-        # add all files in the artifacts folder
-        find artifacts -type f -exec gh release upload $product_version {} \;
+      if: ${{ inputs.deploy == 'true' }}
+      uses: softprops/action-gh-release@v1
+      with:
+        prerelease: true
+        name: "Release of ${{ steps.vars-after-raise.outputs.pkg_version  }}"
+        generate_release_notes: true
+        files: artifacts/*.deb
+        tag_name: ${{ steps.vars-after-raise.outputs.pkg_version  }}
+        target_commitish: ${{ steps.commit.outputs.commit_id }}
       env:
         GITHUB_TOKEN: ${{ secrets.gh_token }}
+    
+    - uses: DenverCoder1/doxygen-github-pages-action@v1
+      if: ${{ inputs.with_docs == 'true' }}
+      with:
+        github_token: ${{ secrets.gh_token }}
+
+  Publish-To-Testing:
+    needs: [Pack]
+    if: ${{ inputs.release == 'false' && inputs.deploy == 'true' }}
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
```

Comment:
avoid naming objects by their undelaying solution, prefer: `artifact_repo_ppas` instead of `nexus_ppa`